### PR TITLE
Add GlobFollowSymlinks method, fixes #11

### LINKED ISSUE #11: Fastwalk should traverse directory symlinks and return all their contents
Consider the following directory structure:

```
/root/
    symlink1   --> folder1
    folder1/
        file1
```

### Current behavior

Currently, the code `zglob.Glob("/root/**/*")` traverses the following entries via fastwalk:

```
/root
/root/symlink1
/root/folder1
/root/folder1/file1
```

And thus there is only one result of the `Glob`: `/root/folder1/file1`

### Expected behavior

What I would expect instead is to get is the following traversal:

```
/root
/root/symlink1
/root/symlink1/file1   <-----------------------------
/root/folder1
/root/folder1/file1
```

And thus I would expect `Glob` to return two results: `/root/folder1/file1` and `/root/symlink1/file1`

### Scenario

We use this library over at [browserpass](https://github.com/dannyvankooten/browserpass/). Specifically, we `Glob` all `**/*.gpg` files and display them in a list. 

Generally with `pass`, the users are encouraged to create directory symlinks in case passwords are shared among several domains, because that is how we could prevent the fishing attacks.

However, due to this bug the symlinked folders are not being traversed, and we cannot prevent the fishing attacks.

Our tracking issue: https://github.com/dannyvankooten/browserpass/issues/85

### PR DESCRIPTION
**Pull request by maximbaz**
@mattn what do you think about this? 

Yes it does not prevent cycles, but it is a rather special scenario which I hope can be dealt separately - none of the browserpass users in #11 will have cycles.

Fixes #11
Closes #12

### CODE REVIEW

**Comment by mattn** [comment]
Thanks your PR. And sorry my long delay. I'll look this soon
.

File: zglob_test.go [hunk 1]
```diff
@@ -136,3 +136,34 @@ func TestMatch(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Chdir(curdir)
+
+	got, err := GlobFollowSymlinks("**/*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected := []string{"foo", "foo/baz.txt", "bar/baz.txt"}
```
Comment (mattn) [inline zglob_test.go]:
```go
	os.MkdirAll(filepath.Join(tmpdir, "foo"), 0755)
	ioutil.WriteFile(filepath.Join(tmpdir, "foo/baz.txt"), []byte{}, 0644)
	defer os.RemoveAll(tmpdir)

	err = os.Symlink(filepath.Join(tmpdir, "foo"), filepath.Join(tmpdir, "bar"))
	if err != nil {
		t.Skip(err.Error())
	}
```
Windows require privilege to make symlink.

File: zglob_test.go [hunk 2]
```diff
@@ -136,3 +136,34 @@ func TestMatch(t *testing.T) {
...
+
+func TestFollowSymlinks(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "zglob")
+	if err != nil {
+		t.Fatal(err)
+	}
+	os.MkdirAll(filepath.Join(tmpdir, "foo"), 0755)
+	ioutil.WriteFile(filepath.Join(tmpdir, "foo/baz.txt"), []byte{}, 0644)
+	os.Symlink(filepath.Join(tmpdir, "foo"), filepath.Join(tmpdir, "bar"))
+	defer os.RemoveAll(tmpdir)
```
Comment (mattn) [inline zglob_test.go]:
```go
	os.MkdirAll(filepath.Join(tmpdir, "foo"), 0755)
	ioutil.WriteFile(filepath.Join(tmpdir, "foo/baz.txt"), []byte{}, 0644)
	defer os.RemoveAll(tmpdir)

	err = os.Symlink(filepath.Join(tmpdir, "foo"), filepath.Join(tmpdir, "bar"))
	if err != nil {
		t.Skip(err.Error())
	}
```
Windows require privilege to make symlink.

Comment (maximbaz) [reply zglob_test.go]: ah, makes sense, thanks for the snippet

**Comment by maximbaz** [comment]
Any further feedback @mattn? Sorry to ping you so often, I'm just eager to get this merged 😄

**Comment by mattn** [comment]
Thanks!

**Pull request merged**